### ISSUE TITLE
Colon characters in key names do not work

### ISSUE THREAD

**Issue by mattnewell** [stelligent/config-lint#126]
From Andy Zietlow:

We are wondering how to handle colon characters in key names like you'll see in the attached example (the file has an actual question in it).
The 2nd example in the file is another rule that was working in tf11 but is now triggering warnings on tf12. We're seeing a lot of similar warnings now with tagging rules that didn't trigger in tf11. Thought I'd pass that info along with the colon-key question just to keep things together but it's also not a critical issue.

[Example.txt](https://github.com/stelligent/config-lint/files/4221097/Example.txt)

**Comment by milldr** [comment]
I am able to recreate this. The terraform resource is handling a colon correctly, but the rule isn't. I've added tests that recreate the issue and am continuing to try to find a solution or work-around. See the tagged branch to be pushed shortly

**Comment by milldr** [comment]
Turns out this is not a bug and was in fact a mistake in the rule itself. To use special characters in a key name, you must include quotes around the key and only the key.

For example:
```
  - id: "CHECK_FOR_COLON"
    message: "Testing for key with a colon"
    resources:
      - aws_s3_bucket_policy
    severity: FAILURE
    category: resource
    assertions:
      - some:
          key: policy.Statement[]
          expressions:
            - key: Condition.Bool."aws:SecureTransport"
              op: eq
              value: false
```
